sources/ClangSharp/Cursors/Decls/TagDecl.cs

```diff
@@ -0,0 +1,27 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using ClangSharp.Interop;
+
+namespace ClangSharp
+{
+    public unsafe class TagDecl : TypeDecl, IDeclContext, IRedeclarable<TagDecl>
```

Comment:
There are interfaces matching the C++ type hierarchy for `DeclContext`, `IRedeclarable`, and `IMergeable`. The latter two are currently just marker interfaces (effectively), but I think I can expose the relevant semantics properly in a later PR.